clients/src/test/java/org/apache/kafka/clients/consumer/internals/ShareCompletedFetchTest.java

```diff
@@ -191,6 +192,69 @@ public void testCorruptedMessage() {
         }
     }
 
+    @Test
+    public void testAcquiredRecords() {
+        long fetchOffset = 5;
+        int startingOffset = 0;
+        int numRecords = 10;        // Records for 0-9
+
+        // Acquiring records 0-2 and 6-8
+        List<ShareFetchResponseData.AcquiredRecords> acquiredRecords = new ArrayList<>(acquiredRecords(0, 3));
+        acquiredRecords.addAll(acquiredRecords(6, 3));
+        ShareFetchResponseData.PartitionData partitionData = new ShareFetchResponseData.PartitionData()
+                .setRecords(newRecords(startingOffset, numRecords, fetchOffset))
+                .setAcquiredRecords(acquiredRecords);
+
+        Deserializers<String, String> deserializers = newStringDeserializers();
+
+        ShareCompletedFetch completedFetch = newShareCompletedFetch(partitionData);
+
+        List<ConsumerRecord<String, String>> records = completedFetch.fetchRecords(deserializers, 10, true).getInFlightRecords();
+        assertEquals(6, records.size());
+        // The first offset should be 0
+        ConsumerRecord<String, String> record = records.get(0);
+        assertEquals(0, record.offset());
+        // The third offset should be 6
+        record = records.get(3);
+        assertEquals(6, record.offset());
+
+        records = completedFetch.fetchRecords(deserializers, 10, true).getInFlightRecords();
+        assertEquals(0, records.size());
+    }
+
+    @Test
+    public void testAcquireOddRecords() {
+        long fetchOffset = 5;
+        int startingOffset = 0;
+        int numRecords = 10;        // Records for 0-9
+
+        // Acquiring all odd Records
+        List<ShareFetchResponseData.AcquiredRecords> acquiredRecords = new ArrayList<>();
+        for (int i = 1; i <= 9; i += 2) {
+            acquiredRecords.add(acquiredRecords(i, 1).get(0));
+        }
+
+        ShareFetchResponseData.PartitionData partitionData = new ShareFetchResponseData.PartitionData()
+                .setRecords(newRecords(startingOffset, numRecords, fetchOffset))
+                .setAcquiredRecords(acquiredRecords);
+
+        Deserializers<String, String> deserializers = newStringDeserializers();
+
+        ShareCompletedFetch completedFetch = newShareCompletedFetch(partitionData);
+
+        List<ConsumerRecord<String, String>> records = completedFetch.fetchRecords(deserializers, 10, true).getInFlightRecords();
+        assertEquals(5, records.size());
+        // The first offset should be 1
+        ConsumerRecord<String, String> record = records.get(0);
+        assertEquals(1, record.offset());
```

Comment:
Really, offsets are long typed, so this should be 1L. I know this is a small point, but I personally find it helpful to follow the convention rigidly for tests because you can tell at a glance which numbers are offsets.